waiter/src/waiter/util/utils.clj

```diff
@@ -441,11 +441,13 @@
   "Creates a component based on the specified :kind"
   [{:keys [kind] :as component-config} & {:keys [context]}]
   (log/info "component:" kind "with config" component-config (if context (str "and context " context) ""))
-  (let [kind-config (get component-config kind)
-        factory-fn (:factory-fn kind-config)]
+  (let [config (if kind
```

Comment:
I think we should not change this and have consistency in how we use kind to create components.

Reply:
Adding *another* sub-map just for the sake of inserting `:kind` seems very cumbersome. I don't like it, but I don't like this either. I'll change it to use `:kind` for now.